### PR TITLE
[Cart][Inventory] Fix cart inventory validation

### PR DESCRIPTION
**Pull request by Arminek**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets | https://github.com/Sylius/Sylius-Standard/issues/135 |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Behat/Context/Ui/Shop/CartContext.php [hunk 1]
```diff
@@ -293,7 +293,7 @@ public function iAddQuantityOfProductsToTheCart($quantity, ProductInterface $pro
 
     /**
      * @Given /^I have(?:| added) (\d+) (products "([^"]+)") (?:to|in) the cart$/
-     * @When /^I add(?:|ed) (\d+) (products "([^"]+)") to the cart$/
+     * @When /^I(?:| again) add(?:|ed) (\d+) (products "([^"]+)") to the cart$/
```
Comment (pamil) [inline src/Sylius/Behat/Context/Ui/Shop/CartContext.php]: This syntax just does not look right :<

File: src/Sylius/Bundle/CoreBundle/Validator/Constraints/InStockValidator.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    private $cartContext;
+
+    /**
+     * @param AvailabilityCheckerInterface $availabilityChecker
+     * @param CartContextInterface $cartContext
+     */
+    public function __construct(AvailabilityCheckerInterface $availabilityChecker, CartContextInterface $cartContext)
+    {
+        $this->availabilityChecker = $availabilityChecker;
+        $this->cartContext = $cartContext;
```
Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Validator/Constraints/InStockValidator.php]: Why?

Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Validator/Constraints/InStockValidator.php]: Just use `$orderItem->getOrder()`.

Comment (Arminek) [inline src/Sylius/Bundle/CoreBundle/Validator/Constraints/InStockValidator.php]: In this case ``Order`` has not been assigned yet.

File: src/Sylius/Bundle/CoreBundle/Resources/config/services/form.xml [hunk 3]
```diff
@@ -165,5 +165,11 @@
             <argument type="service" id="form.registry" />
             <tag name="form.type" alias="sylius_product_variant" />
         </service>
+
+        <service id="sylius.form.type.add_cart_item" class="Sylius\Bundle\CoreBundle\Form\Type\Order\AddCartItemType">
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/CoreBundle/Resources/config/services/form.xml]: Let's move this to OrderBundle.

File: src/Sylius/Bundle/ShopBundle/Resources/config/routing/cart.yml [hunk 4]
```diff
@@ -45,7 +45,7 @@ sylius_shop_cart_item_add:
                 method: createForProduct
                 arguments: [expr:service('sylius.repository.product').find($productId)]
             form:
-                type: sylius_cart_item
+                type: sylius_add_cart_item
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/ShopBundle/Resources/config/routing/cart.yml]: sylius_add_to_cart :)

**Comment by michalmarcinkowski** [comment]
Does it also fix https://github.com/Sylius/Sylius/issues/6635?

**Comment by michalmarcinkowski** [comment]
Great job Arek! 👍

**Pull request merged**